ql/src/java/org/apache/hadoop/hive/ql/udf/generic/UnixTimeFormatter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.udf.generic;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.common.type.TimestampTZUtil;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import java.time.ZoneId;
+import java.util.concurrent.ExecutionException;
+
+public abstract class UnixTimeFormatter<T> {
+
+  enum Type {
+    SIMPLE {
+      @Override
+      UnixTimeFormatter<?> newFormatter(ZoneId zone) {
```

Comment:
nit:  add final for the zone to be consistent with the DATETIME newFormatter method.

Reply:
Instead of adding, I opted to remove the final (just adds noise in this case) from the other method. https://github.com/apache/hive/pull/4615/commits/d8e02d58895613197aa3190e3c64c890a601228f